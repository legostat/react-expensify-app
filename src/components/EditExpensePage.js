import React from "react";
import { connect } from "react-redux";
import Modal from "react-modal";
import FontAwesomeIcon from "@fortawesome/react-fontawesome";
import faTrash from "@fortawesome/fontawesome-free-solid/faTrashAlt";
import ExpenseForm from "./ExpenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

Modal.setAppElement("#app");

export class EditExpensePage extends React.Component {
  constructor() {
    super();
    this.state = {
      modalIsOpen: false
    };
  }
  handleOpenModal = () => {
    this.setState({ modalIsOpen: true });
  };
  handleCloseModal = () => {
    this.setState({ modalIsOpen: false });
  };
  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push("/");
  };
  onRemoveExpense = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  };
  render() {
    return (
      <React.Fragment>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">Edit Expense</h2>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button
            className="button button--secondary"
            onClick={this.handleOpenModal}
          >
            <FontAwesomeIcon icon={faTrash} />
            Remove Expense
          </button>
          <Modal
            isOpen={this.state.modalIsOpen}
            onRequestClose={this.handleCloseModal}
            contentLabel="Remove Expense"
            closeTimeoutMS={200}
            className="modal"
            overlayClassName="modal-overlay"
          >
            <h3 className="modal__title">
              Do you realy want to remove expense?
            </h3>
            <div className="modal__body">
              <button className="button" onClick={this.onRemoveExpense}>
                Yes
              </button>
              <button
                className="button button--secondary"
                onClick={this.handleCloseModal}
              >
                No
              </button>
            </div>
          </Modal>
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

// eslint-disable-next-line no-unused-vars
const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: data => dispatch(startRemoveExpense(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
