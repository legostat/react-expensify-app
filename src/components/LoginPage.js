import React from 'react';
import {connect} from 'react-redux';
import {startLoginWithGoogle, startLoginWithFacebook, startLoginWithTwitter} from '../actions/auth';

export const LoginPage = ({startLoginWithGoogle, startLoginWithFacebook, startLoginWithTwitter}) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p className="box-layout__text">It's time to get your expenses under control.</p>
      <button className="button" onClick={startLoginWithGoogle}>Login with Google</button>
      <button className="button" onClick={startLoginWithFacebook}>Login with Facebook</button>
      <button className="button" onClick={startLoginWithTwitter}>Login with Twitter</button>
    </div>    
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
  startLoginWithFacebook: () => dispatch(startLoginWithFacebook()),
  startLoginWithTwitter: () => dispatch(startLoginWithTwitter())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);