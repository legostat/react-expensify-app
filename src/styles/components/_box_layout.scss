.box-layout {
  background: url('/images/bg.jpg');
  background-size: cover, cover;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;

  &__box {
    background: fade-out(#fff, .15);
    width: 30rem;
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    box-shadow: 2px 5px 15px black;
  }

  &__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
  }

  &__buttons {
    display: flex;
    flex-direction: column;
    align-items: stretch;
  }
}