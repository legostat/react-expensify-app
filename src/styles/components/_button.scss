.button {
  display: inline-block;
  background: $blue;
  border: none;
  border-radius: 5px;
  box-shadow: 2px 5px 12px rgba(0, 0, 0, .5);   
  color: #fff;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  transition: all .2s;
  &:hover {
    box-shadow: 1px 3px 8px rgba(0, 0, 0, .6);
    background: lighten($blue, 8%);
    color: #fefefe;
  }

  &--link {
    background: none;
    box-shadow: none;
    &:hover {
      background: none;
      box-shadow: none;
      color: #efefef;
    }
  }

  &--secondary {
    background: $grey;
    &:hover {
      background: lighten($grey, 8%);
      color: #fefefe;
    }
  }

  &--login {
    padding-left: 2.5rem;
    text-align: left;
    margin: $s-size .5rem;    
  }
}