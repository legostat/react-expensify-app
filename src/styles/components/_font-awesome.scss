.svg-inline--fa {
  margin-right: 1rem;
}