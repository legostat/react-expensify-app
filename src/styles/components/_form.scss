.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }

  &__error{
    margin: 0 0 $m-size 0;
    font-style: italic;
  }

  &__group {
    display: flex;
    flex-direction: column;    

    & > input {
      margin-bottom: $m-size;
    }
    
    @media (min-width: $desktop-breakpoint - 15rem) {
      flex-direction: row;    

      & > input {
        flex-grow: 1;
        margin-right: $m-size;
        margin-bottom: 0;
      }
    }
  }
}