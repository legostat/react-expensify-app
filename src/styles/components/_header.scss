.header {
  background-color: $dark-blue;

  &__content {
    padding: $s-size 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__title {
    color: #fff;
    text-decoration: none;

    h1 {
      margin: 0;
    }
  }
}