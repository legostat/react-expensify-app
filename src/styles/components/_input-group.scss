.input-group{
  margin-bottom: $m-size;
  display: flex;
  flex-direction: column;  
    
  &__item {
    margin-bottom: $s-size;    
  }

  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
    flex-direction: row;

    &__item {
      margin: 0 $s-size 0 0;
    }
  }
}