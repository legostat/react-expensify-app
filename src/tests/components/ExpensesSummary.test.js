import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with 2 expenses and total amount $94.34', () => {
  const wrapper = shallow(<ExpensesSummary expensesTotal={9434} expenseCount={2}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 1 expense and total amount $94.34', () => {
  const wrapper = shallow(<ExpensesSummary expensesTotal={9434} expenseCount={1}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with 0 expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesTotal={0} expenseCount={0}/>);
  expect(wrapper).toMatchSnapshot();
});