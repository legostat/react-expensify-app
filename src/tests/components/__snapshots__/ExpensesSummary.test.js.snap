// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 0 expenses 1`] = `
<div>
  <h2>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h2>
</div>
`;

exports[`should render ExpensesSummary with 1 expense and total amount $94.34 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    $94.34
  </h2>
</div>
`;

exports[`should render ExpensesSummary with 2 expenses and total amount $94.34 1`] = `
<div>
  <h2>
    Viewing 
    2
     
    expenses
     totalling 
    $94.34
  </h2>
</div>
`;
